src/Test/TestCases.Runtime/ParallelBranchTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using Shouldly;
+using System;
+using System.Activities;
+using System.Activities.ParallelTracking;
+using System.Linq;
+using WorkflowApplicationTestExtensions;
+using WorkflowApplicationTestExtensions.Persistence;
+using Xunit;
+
+namespace TestCases.Runtime;
+
+
+public class ParallelBranchTestsJson : ParallelBranchTests
+{
+    protected override IWorkflowSerializer Serializer => new JsonWorkflowSerializer();
+}
+
+public class ParallelBranchTests
+{
+    public class TestCodeActivity(Action<CodeActivityContext> onExecute) : CodeActivity
+    {
+        protected override void Execute(CodeActivityContext context)
+        {
+            onExecute(context);
+        }
+    }
+
+    protected virtual IWorkflowSerializer Serializer => new DataContractWorkflowSerializer();
+    string _parentLevel = default;
+    readonly string _noBranch = default;
+
+    private void Run(params Action<CodeActivityContext>[] onExecute)
+    {
+        new WorkflowApplication(new SuspendingWrapper(onExecute.Select(c => new TestCodeActivity(c))))
+        {
+            InstanceStore = new MemoryInstanceStore(Serializer)
+        }.RunUntilCompletion();
+    }
+
+    private void RunWithParent(params Action<CodeActivityContext>[] onExecute)
+    {
+        Run([SetParent, .. onExecute]);
+        void SetParent(CodeActivityContext context)
+        {
+            var parent = context.CurrentInstance.Parent;
+            parent.GetCurrentParallelBranchId().ShouldBe(_noBranch);
+            parent.MarkNewParallelBranch();
```

Comment:
We are setting the parent Id with the child running.  Even though this works correctly I think (and the child sees the ExecutionProperty even though we live-edited the parent), I think we should do it in the order the activities do.

Reply:
?